apps/vs-code-designer/src/app/commands/deploy/hybridLogicApp/index.ts

```diff
@@ -88,6 +101,157 @@ export const deployHybridLogicApp = async (context: IActionContext, node: SlotTr
+  }
+};
+
+const waitForContainerAppProvisioning = async (fqdn: string): Promise<void> => {
+  const maxRetries = 40; // Maximum number of retries
+  const delay = 3000; // Delay between retries in milliseconds (3 seconds)
+  const zipDeployUrl = `https://${fqdn}/api/zipDeploy`;
+
+  for (let attempt = 0; attempt < maxRetries; attempt++) {
+    try {
+      const response = await axios.get(zipDeployUrl, { timeout: 3000 });
+      if (response.status === 200) {
+        return; // Container app is ready
+      }
+    } catch (_) {
+      // Ignore errors and retry
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, delay)); // Wait before retrying
+  }
+
+  throw new Error('Logic app is not ready to serve traffic after waiting.');
+};
+
+const waitForIngressFqdn = async (node: SlotTreeItem, clientContainer: any, maxRetries = 30, delay = 3000): Promise<void> => {
+  for (let attempt = 0; attempt < maxRetries; attempt++) {
+    if (node.hybridSite.configuration?.ingress?.fqdn !== null) {
+      return; // FQDN is available
+    }
+
+    try {
+      // Refresh the hybrid site details
+      node.hybridSite = await clientContainer.containerApps.get(node.resourceGroupName, node.hybridSite.name);
+    } catch (error) {
+      console.error(`Error refreshing hybrid site details: ${error.message}`);
+    }
+
+    // Wait before retrying
+    await new Promise((resolve) => setTimeout(resolve, delay));
+  }
+
+  throw new Error('Failed to retrieve ingress FQDN after waiting.');
+};
+
+const createZipFileOnDisk = async (sourceDir: string): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const tempDir = os.tmpdir();
+    const zipFilePath = path.join(tempDir, `logicapp-deploy-${Date.now()}.zip`);
+    const zipFile = new yazl.ZipFile();
+
+    // Add all files in the directory to the zip
+    const addDirectoryToZip = (dir: string, basePath: string) => {
+      const files = fs.readdirSync(dir);
+      for (const file of files) {
+        const fullPath = path.join(dir, file);
+        const relativePath = path.relative(basePath, fullPath);
+
+        if (fs.statSync(fullPath).isDirectory()) {
+          addDirectoryToZip(fullPath, basePath); // Recursively add subdirectories
+        } else {
+          zipFile.addFile(fullPath, relativePath); // Add file to the zip
+        }
+      }
+    };
+
+    addDirectoryToZip(sourceDir, sourceDir);
+
+    // Write the zip file to disk
+    const output = fs.createWriteStream(zipFilePath);
+    zipFile.outputStream.pipe(output);
+
+    output.on('close', () => resolve(zipFilePath));
+    output.on('error', (err) => reject(err));
+    zipFile.end();
+  });
+};
+
+const getAccessTokenForZipDeploy = async (node: SlotTreeItem, context: ILogicAppWizardContext): Promise<string> => {
+  // Retrieve environment variables for AAD
+  const tenantId = node.hybridSite.template.containers[0]?.env.find((e) => e.name === workflowAppAADTenantId)?.value;
```

Comment:
Can you store the `node.hybridSite.template.containers[0]?.env` in a constant? since you are accessing in multiple times. 
And also add safe access to the array like `.containers?.[0]?.env`